app/models/manageiq/providers/cisco_intersight/physical_infra_manager/event_catcher/stream.rb

```diff
@@ -18,19 +18,31 @@ def stop
       @stop_polling = true
     end
 
-    def poll
-      @ems.with_provider_connection do |connection|
-        catch(:stop_polling) do
-          loop do
-            connection.events.each do |activity|
-              throw :stop_polling if @stop_polling
-              yield activity.to_h
-            end
-            sleep @poll_sleep
+    def poll(&block)
+      since = Time.new(2000).utc.iso8601
```

Comment:
Sorry @agrare, we would also like fetch historical events for the events and with https://github.com/ManageIQ/manageiq-providers-cisco_intersight/pull/55/commits/4b93ca66016e6a99e5ca70424c325c404e8bb4bf only events that come after creating the provider get fetched. Any suggestions how to get historical events also? Thanks!

Reply:
Yeah so this is why I was suggestion to check the most recent event in  `ems.ems_events` so if we were shutdown for e.g. an hour we could use that as the starting point on the next startup.  If there are no events (aka first run ever) then you can go back as far as you want on the initial run.

Something like `since = ems.ems_events.order(:timestamp).pluck(:timestamp).last&.utc&.iso8601 || 1.month.ago.utc.iso8601` might do the trick?

Reply:
Thanks Adam. Will create a follow-up PR that will fix this one.